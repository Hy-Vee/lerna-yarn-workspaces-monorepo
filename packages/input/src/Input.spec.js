import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Input from '.';

describe('Input', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<Input />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('placeholder', () => {
        const tree = renderer.create(<Input placeholder="test" />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('disabled', () => {
        const tree = renderer.create(<Input disabled />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
