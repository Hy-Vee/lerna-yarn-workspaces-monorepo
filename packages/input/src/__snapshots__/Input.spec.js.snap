// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input disabled 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  color: #868686;
  font-family: system-ui,-apple-system,'Segoe UI',Roboto,'Helvetica Neue',Arial;
  font-size: 0.8rem;
  text-transform: uppercase;
}

.c2 {
  background-color: transparent;
  border-radius: 4px;
  border: 1px solid #ccc;
  box-shadow: none;
  box-sizing: border-box;
  caret-color: #000;
  color: #000;
  display: block;
  font-size: 14px;
  height: 40px;
  line-height: 26px;
  padding: 10px;
  margin: 4px 0 8px;
  width: 100%;
  color: #999999;
  background-color: #fafafa;
  cursor: not-allowed;
}

.c2::-webkit-input-placeholder {
  color: #999;
}

.c2::-moz-placeholder {
  color: #999;
}

.c2:-ms-input-placeholder {
  color: #999;
}

.c2::placeholder {
  color: #999;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="email"
  >
    Email Address
  </label>
  <input
    className="c2"
    disabled={true}
    id="email"
    type="text"
  />
</div>
`;

exports[`Input placeholder 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  color: #868686;
  font-family: system-ui,-apple-system,'Segoe UI',Roboto,'Helvetica Neue',Arial;
  font-size: 0.8rem;
  text-transform: uppercase;
}

.c2 {
  background-color: transparent;
  border-radius: 4px;
  border: 1px solid #ccc;
  box-shadow: none;
  box-sizing: border-box;
  caret-color: #000;
  color: #000;
  display: block;
  font-size: 14px;
  height: 40px;
  line-height: 26px;
  padding: 10px;
  margin: 4px 0 8px;
  width: 100%;
}

.c2::-webkit-input-placeholder {
  color: #999;
}

.c2::-moz-placeholder {
  color: #999;
}

.c2:-ms-input-placeholder {
  color: #999;
}

.c2::placeholder {
  color: #999;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="email"
  >
    Email Address
  </label>
  <input
    className="c2"
    disabled={false}
    id="email"
    placeholder="user@gmail.com"
    type="text"
  />
</div>
`;

exports[`Input renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  color: #868686;
  font-family: system-ui,-apple-system,'Segoe UI',Roboto,'Helvetica Neue',Arial;
  font-size: 0.8rem;
  text-transform: uppercase;
}

.c2 {
  background-color: transparent;
  border-radius: 4px;
  border: 1px solid #ccc;
  box-shadow: none;
  box-sizing: border-box;
  caret-color: #000;
  color: #000;
  display: block;
  font-size: 14px;
  height: 40px;
  line-height: 26px;
  padding: 10px;
  margin: 4px 0 8px;
  width: 100%;
}

.c2::-webkit-input-placeholder {
  color: #999;
}

.c2::-moz-placeholder {
  color: #999;
}

.c2:-ms-input-placeholder {
  color: #999;
}

.c2::placeholder {
  color: #999;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="email"
  >
    Email Address
  </label>
  <input
    className="c2"
    disabled={false}
    id="email"
    type="text"
  />
</div>
`;
